RuleDocumentation/PossibleIncorrectUsageOAssignmentOperator.md

```diff
@@ -0,0 +1,54 @@
+# PossibleIncorrectUsageOfAssignmentOperator
+
+**Severity Level: Information**
+
+## Description
+
+In many programming languages, the equality operator is denoted as `==` or `=` in many programming languages, but `PowerShell` uses `-eq`. Therefore it can easily happen that the wrong operator is used unintentionally and this rule catches a few special cases where the likelihood of that is quite high.
+
+The rule looks for usages of `==` and `=` operators inside `if`, `else if`, `while` and `do-while` statements but it will not warn if any kind of command or expression is used at the right hand side as this is probably by design.
+
+## Example
+
+### Wrong
+
+```` PowerShell
+if ($a = $b)
+{
+    ...
+}
+````
+
+```` PowerShell
+if ($a == $b)
+{
+
+}
+````
+
+### Correct
+
+```` PowerShell
+if ($a -eq $b) # Compare $a with $b
+{
+    ...
+}
+````
+
+```` PowerShell
+if ($a = Get-Something) # Only execute action if command returns something and assign result to variable
```

Comment:
I know what you mean but it was a strong desire of the community to not see warnings when assignment is by design as this is a common use case. See referenced issue and there was also a discussion on the testing channel in Slack. I can sort of understand this fear/annoyance of people about false positives, especially given that one cannot suppress on a per-line basis in PSSA. I guess it is better to warn in most cases and help save the developer time rather than always warning and annoying the developer.